base/abstractarraymath.jl

```diff
@@ -92,6 +92,10 @@ imag{T<:Real}(x::AbstractArray{T}) = zero(x)
 +{T<:Number}(x::AbstractArray{T}) = x
 *{T<:Number}(x::AbstractArray{T,2}) = x
 
+function lerpi(j::Integer, d::Integer, A::AbstractArray, B::AbstractArray)
```

Comment:
With this implementation,
```jl
r = linspace(rand(3), rand(3), 10^5)
julia> @time mysum(r)
  0.021307 seconds (200.13 k allocations: 21.370 MB, 24.71% gc time)
3-element Array{Float64,1}:
 29495.6
 62857.1
 50091.0
```
If I change `unsafe_getindex` to `lerpi.(i-1, r.lendiv, r.start, r.stop)` I get
```jl
julia> @time mysum(r)
  1.134825 seconds (3.40 M allocations: 161.736 MB, 2.15% gc time)
3-element Array{Float64,1}:
 29495.6
 62857.1
 50091.0
```
which is too much of a performance hit. Obviously we could fix that but I'd prefer to make that a separate PR.